Checks default applications

dasantiago opened the pull request:
It will check if some specific mimetypes will open with some specific applications
by using the gvfs-mime utility.
This needs to be done with gvfs-mime, instead of comparing needles, because it some mimetypes are registered with console applications that don't show any window.

coolo commented:
but we keep needles for those that do?

dasantiago commented:
@coolo There are no change in the needles. The working needles, will keep continue to work if the application is correctly associated to the mimetypes.

coolo commented on tests/x11regressions/gnomecase/gnome_default_applications.pm:
```diff
@@ -19,55 +19,82 @@ sub run {
-    # Clean the test directory
-    x11_start_program("rm -rf /home/$username/gnometest");
+    my $return = 0;
+    for my $app (@apps) {
```
what's the point of a loop if you return from the first item?

coolo commented on tests/x11regressions/gnomecase/gnome_default_applications.pm:
```diff
@@ -19,55 +19,82 @@ sub run {
+        assert_script_run "cp /usr/share/w3m/w3mhelp.html /home/$username/gnometest/";
+        assert_script_run "tar cjvf /home/$username/gnometest/test.tar.bz2 -C /home/$username/gnometest/ test.pdf";
+        assert_script_run "tar czvf /home/$username/gnometest/test.tar.gz -C /home/$username/gnometest/ test.pdf";
+        send_key 'alt-f4';
```
to avoid indenting everything I suggest to put the old code into a function called from within here

okurz commented on tests/x11regressions/gnomecase/gnome_default_applications.pm:
```diff
@@ -19,55 +19,82 @@ sub run {
+            ['image/png',           'eog.desktop'],
+            ['application/pdf',     'evince.desktop'],
+            ['application/x-bzip2', 'org.gnome.FileRoller.desktop'],
+            ['application/gzip',    'org.gnome.FileRoller.desktop']))
```
IMHO that's not very readable. How about you call that before the if, assign it to a boolean value and check that?

coolo replied:
yeah. as you want to loop over it anyway, you can do

```
my $failed = 0;
$failed ||= check_default('image/png', 'eog.desktop');
$failed ||= ...

if ($failed) {...
```

dasantiago commented:
New job with the recommended changes:
http://tragicbox.suse.cz/tests/42#step/gnome_default_applications/16

@okurz @coolo Is the code better this way?

coolo commented:
Can you also run it for sp3 so we see it working still?

okurz commented:
The commit message could have a bit more details, e.g. what issue is fixed, why this needs to change, to which versions of which product it applies, link to a bug or a fate entry, the choices you made, etc. Also see http://chris.beams.io/posts/git-commit/ as a helpful guide how to write good commit messages.

okurz commented on tests/x11regressions/gnomecase/gnome_default_applications.pm:
```diff
@@ -20,23 +20,48 @@ sub run {
     send_key "ctrl-l";
     type_string "/home/$username/gnometest\n";
     send_key "ret";
+    sleep 5;
```
Why should we need a sleep 5 in front of an assert_screen? I don't think that is needed. Did you hit a problem why you added this?

dasantiago replied:
That sleep was already in the code. I didn't changed it. I'll test without it.

dasantiago commented:
@okurz 
The commit message contains the poo number.

I also explained why i used a command line and not a bunch of needles... Like for example, i could check for the eog needle and so on... But for example, for archives, the default app was the "Disk Image Mounter" or something like this. This puzzled me while i was looking to the job images, because i wasn't seeing any window being shown, which raised some question (is the application being invoked?  Was the application invoked and it exploded? What's the application being invoked? Etc...)  so i decided to use the gvfs-mime utility.

Some of the points are valid, but in terms of new information, i believe i have nothing to add.

dasantiago commented:
@coolo 

SP2: http://tragicbox.suse.cz/tests/48#step/gnome_default_applications/25
SP3: http://tragicbox.suse.cz/tests/49#step/gnome_default_applications/59

@okurz 

Removed the sleep. No difference whatsoever in the execution.
Also, changed a bit the commit message.

coolo commented:
I like it now :)

okurz commented:
thanks for attending the comments
